lib/class/api.class.php

```diff
@@ -560,7 +790,16 @@ public static function url_to_song($input)
         // Don't scrub, the function needs her raw and juicy
         $data = Stream_URL::parse($input['url']);
         ob_end_clean();
-        echo XML_Data::songs(array($data['id']));
+
+        //Whatever format the user wants
+        $outputFormat = $input['format'];
+
+        if ($outputFormat == 'json') {
+            echo JSON_Data::songs(array($data['id']));
```

Comment:
Same about the array. Will not repeat it in the following of the diff, but it still applies.